code/controllers/subsystem/throwing.dm

```diff
@@ -102,6 +105,12 @@ SUBSYSTEM_DEF(throwing)
 
 	qdel(src)
 
+#define TURF_THROWNTHING_CHECK(varname) \
+	var/atom/##varname = thrownthing.loc;\
+	SEND_SIGNAL(varname, COMSIG_TURF_THROWNTHING_CHECK, src);\
```

Comment:
Pretty whack as well, signals should ideally only be fired from exactly one place. What is this about?